File: code/mathematical_algorithms/src/sum_of_digits/sum_of_digits.java
/*

// mathematical algorithms | sum of digits | sum of digits | JAVA
// Part of Cosmos by OpenGenus Foundation

*/


// Part of Cosmos by OpenGenus Foundation
public class SumOfDigits {
    public static int sumDigits(long n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n = n / 10;
        }
        return sum;
    }
}

File: code/mathematical_algorithms/src/sum_of_digits/sum_of_digits.rb
# mathematical algorithms | sum of digits | sum of digits | RUBY
// Part of Cosmos by OpenGenus Foundation

=end


## Part of Cosmos by OpenGenus Foundation
def sum_of_digits(num)
  num = num.abs
  sum = 0
  while num > 0
    sum += num % 10
    num /= 10
  end
  sum
end
